### PR TITLE
Added support for dependency destination specification. (Resolves #1038)

### LINKED ISSUE #1038: Embed dependencies to specified destination
When I setup dependencies using XcodeGen, it creates a copy files build phase for the dependencies and puts them automatically in Frameworks folder.

In Xcode, there is a combo-box that allows changing the destination. You do define and support these destinations for Copy Phase that works with sources but I have not found a way how to specify it when embedding dependencies.

Could you please add this feature?

### PR DESCRIPTION
**Pull request by JakubBednar**

### CODE REVIEW

**Review by freddi-kit — COMMENTED** [review]
Hello, @JakubBednar. Before submitting a pull request, please add a test to it. Please refer to the test topic in CONTRIBUTING.md https://github.com/yonaskolb/XcodeGen/blob/master/CONTRIBUTING.md#tests

**Comment by JakubBednar** [comment]
Hello, frankly it is harder for me to navigate your unit-tests than to update the actual code and add the feature. :) So I was hopping to get a general feedback on the proposed feature and solution first. If you'd agree that this is something that makes sense to you and the proposed way of implementation is correct as well I will add the tests. But if you decide that you do not want this feature or you want it implemented differently than I would be just wasting my time on the tests.

With that said, I suppose I should extend both ProjectSpecTests and PBXProjGeneratorTests. Or is there anything else?

File: Tests/XcodeGenKitTests/ProjectGeneratorTests.swift [hunk 1]
```diff
@@ -1581,6 +1581,1767 @@ class ProjectGeneratorTests: XCTestCase {
...
+                        name: "appB",
+                        type: .application,
+                        platform: .macOS
+                    )
+                    
+                    $0.it("does not embed them without copy phase spec") {
+                        
+                        // given
+                        let dependencies = [
+                            Dependency(type: .target, reference: appA.name, embed: true),
```
Comment (JakubBednar) [inline Tests/XcodeGenKitTests/ProjectGeneratorTests.swift:1625]: There are several target types that original XcodeGen was not embedding even when embed was set to true. Or maybe it puts them into different sections than copyFiles section? I am not sure and I have not found any tests describing original embedding scheme for different types of dependencies

**Comment by freddi-kit** [comment]
Thank you for your update and I'm sorry for my late reply. Now I became Collaborator for this repo so please give a time to learn and I'll review it soon!

File: Sources/XcodeGenKit/PBXProjGenerator.swift [hunk 2]
```diff
@@ -1127,6 +1168,22 @@ public class PBXProjGenerator {
             buildPhases.append(copyFilesPhase)
         }
 
+        if !customCopyDependenciesReferences.isEmpty {
+            
+            let splitted = splitCopyDepsByDestination(customCopyDependenciesReferences)
+            for (phase, references) in splitted {
+                
+                // I would prefer force unwrap as it should never return nil, but I am not sure about code policy
+                guard let destination = phase.destination.destination else { continue }
```
[the remark this answers is not in the document]
Comment (freddi-kit) [reply Sources/XcodeGenKit/PBXProjGenerator.swift:1177]: FYI: I had a same opinion at here https://github.com/yonaskolb/XcodeGen/pull/796#discussion_r388317066 and it is accepted, so I think what you write is okay :)

**Comment by JakubBednar** [comment]
Hello, @freddi-kit. Anything I can do to make this get merged?

**Comment by jakubkiermasz-zd** [comment]
Hello!
Seems like this resolves #1102, doesn't it :)?

@JakubBednar, @freddi-kit any updates?

**Comment by jakubkiermasz-zd** [comment]
If you are interested, https://github.com/zendesk/XcodeGen is the fork aligned with the develop that we will use till it's merged.

**Comment by yonaskolb** [comment]
Thanks for the great work here @JakubBednar. It all seems good. Only thing is I'm a little hesitant to merge 1700 lines of testing code that will have to be maintained. If you can reduce this a bit with the use of some helper functions or some setup objects that would be great. Once done and the conflicts are resolved, we can get this merged 👍 

@jakubkiermasz-zd are you confirming this solves your particular use case?

**Comment by freddi-kit** [comment]
Ah sorry for not adding comments for a long time, I thought I asked others to review but it looks ask review button in github not works when I tapping, sorry. 🙇

**Comment by jakubkiermasz-zd** [comment]
@yonaskolb I am. It works like a charm ^^

**Comment by JakubBednar** [comment]
I have resolved the conflicts and remove IMO the only boiler-plate code in the unit-tests. Please note the following about the unittests:

1. I did cover all available dependency types and there is a lot of it
2. For each dependency type, the first test actually fixes how xcodegen worked before changes. For some dependency types it ignored `embed: true` for some it ignored `embed: false`. For bundle dependencies it still does as these seem to take different path in the code and I was not sure whether it was an intention or a bug so I did not change it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -66,6 +66,7 @@
 ## 2.20.0
 
 #### Added
+- Allow specifying a `copy` setting for each dependency. [#1038](https://github.com/yonaskolb/XcodeGen/pull/1039) @JakubBednar
```
Comment (yonaskolb) [inline CHANGELOG.md:69]: Can we move this up into `Next Version`

**Review by yonaskolb — CHANGES_REQUESTED** [review]
Great, thank you @JakubBednar

**Pull request merged**